core/src/test/java/io/aiven/klaw/UsersTeamsControllerIT.java

```diff
@@ -670,4 +673,136 @@ public void myProfileInfo() throws Exception {
         .hasSize(2)
         .containsExactlyInAnyOrder(1001, 1002);
   }
+
+  @Test
+  @Order(22)
+  public void registerAndApproveUser() throws Exception {
+    String role = "USER";
+    RegisterUserInfoModel userInfoModel = mockMethods.getRegisterUserInfoModel(user3, role);
+    String jsonReq = OBJECT_MAPPER.writer().writeValueAsString(userInfoModel);
+
+    String registerUserResponse =
+        mvc.perform(
+                MockMvcRequestBuilders.post("/registerUser")
+                    .with(user(superAdmin).password(superAdminPwd))
+                    .content(jsonReq)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    ApiResponse apiResponse =
+        OBJECT_MAPPER.readValue(registerUserResponse, new TypeReference<>() {});
+    assertThat(apiResponse.isSuccess()).isTrue();
+
+    String approveUserResponse =
+        mvc.perform(
+                MockMvcRequestBuilders.post("/execNewUserRequestApprove")
+                    .with(user(superAdmin).password(superAdminPwd))
+                    .param("username", user3)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    apiResponse = OBJECT_MAPPER.readValue(approveUserResponse, new TypeReference<>() {});
+    assertThat(apiResponse.isSuccess()).isTrue();
+
+    String userDetailsResponse =
+        mvc.perform(
+                MockMvcRequestBuilders.get("/getUserDetails")
+                    .with(user(superAdmin).password(superAdminPwd))
+                    .param("userId", user3)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    UserInfoModelResponse userInfoModelActual =
+        new ObjectMapper().readValue(userDetailsResponse, new TypeReference<>() {});
+    assertThat(userInfoModelActual.getTeam()).isEqualTo(INFRATEAM);
+  }
+
+  @Test
+  @Order(22)
+  public void registerAndDeclineAndReRegisterAndApproveUser() throws Exception {
+    String role = "USER";
+    RegisterUserInfoModel userInfoModel = mockMethods.getRegisterUserInfoModel(user4, role);
+    String jsonReq = OBJECT_MAPPER.writer().writeValueAsString(userInfoModel);
+
+    String registerUserResponse =
+        mvc.perform(
```

Comment:
We can probably extract these three performs into a method each and re-use them. 
It would reduce code  duplication